### PR TITLE
zsh modules' files are not recognized

### PR DESCRIPTION
**Pull request by Freed-Wu**
Problem: zsh modules' files are not recognized
Solution: add some patterns for zsh modules' files

### CODE REVIEW

File: runtime/autoload/dist/ft.vim [hunk 1]
```diff
@@ -408,12 +408,16 @@ enddef
 
 # Distinguish between "default", Prolog and Cproto prototype file.
```
Comment (k-takata) [inline runtime/autoload/dist/ft.vim]: This line should be also updated.

File: src/testdir/test_filetype.vim [hunk 2]
```diff
@@ -855,7 +855,7 @@ def s:GetFilenameChecks(): dict<list<string>>
     zsh: ['.zprofile', '/etc/zprofile', '.zfbfmarks', 'file.zsh', 'file.zsh-theme', 'file.zunit',
           '.zcompdump', '.zlogin', '.zlogout', '.zshenv', '.zshrc', '.zsh_history',
           '.zcompdump-file', '.zlog', '.zlog-file', '.zsh', '.zsh-file',
-          'any/etc/zprofile', 'zlog', 'zlog-file', 'zsh', 'zsh-file'],
+          'any/etc/zprofile', 'zlog', 'zlog-file', 'zsh', 'zsh-file', 'file.mmi'],
```
Comment (k-takata) [inline src/testdir/test_filetype.vim]: Mistake of 'file.mdd'?

Comment (Freed-Wu) [reply src/testdir/test_filetype.vim]: Fixed.

File: runtime/autoload/dist/ft.vim (hunk 1 above)
Comment (Freed-Wu) [reply runtime/autoload/dist/ft.vim]: Fixed.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**